src/preamble.js

```diff
@@ -256,6 +256,10 @@ var HEAP,
   HEAP32,
 /** @type {Uint32Array} */
   HEAPU32,
+/** @type {BigInt64Array} */
+  HEAP64,
+/** @type {BigUint64Array} */
+  HEAPU64,
```

Comment:
We are changing the code that writes an `i64` to use these instead of the previous 2 writes to `HEAP32`, which also holds for any `i64`s in wasm32 mode.